### PR TITLE
Finish folding feature and add test framework.

### PR DESCRIPTION
**Pull request by AbnerZheng**
In this PR,  I have finished a new feature- folding. People can fold block document commenter and procedure between "{" and "}". 
And I do add the test support for this IDE, in this ways we can write test case for lexer,parser and some other feature rather than run another editor every time.

### CODE REVIEW

**Comment by tgrospic** [comment]
It's nice to see the folding feature implemented. :smiley: 

Can you please add a section in README file with a short guide how to use tests that you added.

File: src/coop/rchain/lang/Rho.bnf
Comment (tgrospic) [inline src/coop/rchain/lang/Rho.bnf:114]: I think we can relax the syntax and allow optional `Proc` inside block.
`ProcBlock ::= "{" [Proc] "}"`

It will be less disturbing when having unfinished code. And also FOR can be defined with `ProcBlock`.

File: .idea/misc.xml
Comment (tgrospic) [inline .idea/misc.xml:1]: I was not sure if something from `.idea` folder should be checked-in. If it is not, you can remove the whole folder from git.

File: .gitignore
Comment (tgrospic) [inline .gitignore:2]: `.idea/workspace.xml` is not needed anymore.

File: src/coop/rchain/lang/folding/RholangFoldingBuilder.java
Comment (tgrospic) [inline src/coop/rchain/lang/folding/RholangFoldingBuilder.java]: Please convert the file to Kotlin, usualy it is much shorter than Java file. (Idea has the Convert command inside Code menu)

**Review by tgrospic — CHANGES_REQUESTED** [review]
I found that folding is working! Before this PR is merged please see my comments.

File: src/coop/rchain/lang/Rho.bnf
Comment (AbnerZheng) [reply src/coop/rchain/lang/Rho.bnf:114]: Yes, you're right.

File: src/coop/rchain/lang/folding/RholangFoldingBuilder.java
Comment (AbnerZheng) [reply src/coop/rchain/lang/folding/RholangFoldingBuilder.java]: Amazing!

File: .idea/misc.xml
Comment (AbnerZheng) [reply .idea/misc.xml:1]: Sure, I will clean these files.

**Comment by AbnerZheng** [comment]
I have translated all java code into kotlin and add explanation of how to use test framework in README.md.

File: src/coop/rchain/lang/folding/RholangFoldingBuilder.kt
Comment (tgrospic) [inline src/coop/rchain/lang/folding/RholangFoldingBuilder.kt:1]: In the project I included `.editorconfig` file which helps to keep formatting consistent. It is set to indent of 2 spaces. If you install _EditorConfig_ plugin you can just reformat the code.

File: .idea/misc.xml
Comment (tgrospic) [reply .idea/misc.xml:1]: You put this file in `.gitignore` list. Should it be deleted?

**Review by tgrospic — COMMENTED** [review]
Great, you converted all Java files. I just have minor things that are not critical.
After the merge I will update the docs to include info about EditorConfig and also required plugin Grammar-Kit.

File: .idea/misc.xml
Comment (AbnerZheng) [reply .idea/misc.xml:1]: I think it should not be deleted, since it is ignored, anyone could maintain his sdk locally, and will not be commit. I'm sorry that it is hard to download these sdk in China.

File: src/coop/rchain/lang/folding/RholangFoldingBuilder.kt
Comment (AbnerZheng) [reply src/coop/rchain/lang/folding/RholangFoldingBuilder.kt:1]: Yes, I will install this plugin, and get all code formatted.

**Review by tgrospic — APPROVED** [review]

**Comment by tgrospic** [comment]
@AbnerZheng thanks for this PR. You can also be rewarded for your work, here you can find more info.
https://github.com/rchain/Members/blob/master/CONTRIBUTING.md#decentralized-budget-allocation

The new version is uploaded and awaiting JetBrains approval.
https://plugins.jetbrains.com/plugin/9833-rholang/update/42643

**Pull request merged**
